tests/test_checks.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+from django.core import checks
+from django.core.management import base, call_command
+from django.test import SimpleTestCase
+from django.test.utils import override_settings
+
+from corsheaders.checks import check_settings
+from corsheaders.conf import corsheaders_settings
+
+
+class SettingsTests(SimpleTestCase):
+
+    @override_settings(CORS_ALLOW_HEADERS=True)
+    def test_cors_allow_headers_failure(self):
+        errors = check_settings(corsheaders_settings)
```

Comment:
Just realized you're passing `corsheaders_settings` object to `check_settings` here - that's not the right argument. Check functions take a list of app configs as their first argument, and since we're not inspecting that in `check_settings`, the correct 'dummy' argument would be an empty list